### PR TITLE
Extract thread logic from Mopy library

### PR DESCRIPTION
**Pull request by Levis92**

### CODE REVIEW

File: mopy/src/main/java/it/chalmers/digit/Mopy.java [hunk 1]
```diff
@@ -16,38 +16,46 @@
 public final class Mopy implements MopedController {
 
 	private static final String PATH = "/etc/onTruck/python/";
+	private static Mopy instance;
```
Comment (fantomsol) [inline mopy/src/main/java/it/chalmers/digit/Mopy.java]: `instance` is not initialized, thus program does not run.

File: mopy/src/main/java/it/chalmers/digit/Mopy.java [hunk 2]
```diff
@@ -16,38 +16,46 @@
 public final class Mopy implements MopedController {
```
Comment (fantomsol) [inline mopy/src/main/java/it/chalmers/digit/Mopy.java:17]: Mopy should have a private constructor.

**Review by fantomsol — DISMISSED** [review]

File: mopy/src/main/java/it/chalmers/digit/Mopy.java (hunk 2 above)
Comment (Levis92) [reply mopy/src/main/java/it/chalmers/digit/Mopy.java:17]: It has now 😉

File: mopy/src/main/java/it/chalmers/digit/Mopy.java [hunk 3]
```diff
@@ -47,7 +52,7 @@ public static Mopy getInstance() {
 		if (instance == null) {
 			// Synchronize to prevent threads to overlap during creation and
 			// outer check is not synchronized to prevent performance loss
-			synchronized (instance) {
+			synchronized (lock) {
 				if (instance == null) {
 					instance = new Mopy();
```
Comment (hulthe) [inline mopy/src/main/java/it/chalmers/digit/Mopy.java]: `instance` should be instantiated on program startup

**Review by hulthe — CHANGES_REQUESTED** [review]

**Review by hulthe — APPROVED** [review]

**Pull request merged**